samcli/commands/validate/validate.py

```diff
@@ -47,6 +49,16 @@ def do_cli(ctx, template):
     click.secho("{} is a valid SAM Template".format(template), fg='green')
 
 
+def _get_boto_client(profile):
+    # If a profile is passed, instantiate the boto3 client from the session.
+    if profile:
+        # Throws a botocore.exceptions.ProfileNotFound for unknown profiles.
```

Comment:
This will throw a ProfileNotFound to the client, which will show a stack trace. We should catch this exception and throw an exception that Inherits from [UserException](https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/commands/local/cli_common/user_exceptions.py#L8).

[Here](https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/commands/validate/validate.py#L43-L45) is what we do in the validate command when the SAM Translator throws an exception for an invalid template.